### PR TITLE
[UI] Accessibility features in user profile

### PR DESCRIPTION
**Pull request by andersonjeccel**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 

## Description

It adds accessibility features:
- Select whether transitions remain smooth or go instant
- Toggle strong borders around clickable elements and dashed outline style
- Adjust the transparency and blur effects to improve readability or visual clarity
- Toggle the visibility of underlines on links that are adjacent to text

image

https://github.com/user-attachments/assets/2e2ea783-bbbd-4ff2-9a6c-f7d49de8d237

---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Click on the user menu in the header
3. Click to open your profile "Account" 
4. Open the tab Accessibility 

**Testing Smooth transitions**
5. Hover your mouse over the **Save** and **Cancel** buttons to memorize the smooth transition
6. Now set _Smooth transitions_ to **Disabled**
7. Hover again over those buttons, you'll notice their color changes instantly

**Testing borders and outline**
8. Click on the tab Accessibility and memorize how the link looks
9. Set Borders and outline style to **Enabled**
10. Click again: you'll see a dashed border appearing around the text to help you locate the focus points on the screen
11. Open the Account details tab
12. Check how all fields now have a strong border on all sides

**Testing reduced transparency and blur**
13. Go to Contacts
14. Open the actions dropdown menu
15. Select **Export to CSV**
16. See the flash message appearing with blurred background
17. Now go back to your account, in the Accessibility tab
18. Set Reduce transparency and blur to **Reduced**
19. Repeat the export to see

**Testing link underlines**
20. On the Accessibility tab, set Link underlines to **Enabled**
21. Check the links on tables, alerts, accordions, paragraphs and also the links to each tab
22. They all have underlines

### CODE REVIEW

**Comment by LordRembo** [comment]
@andersonjeccel Just an idea but maybe the section title could be 'Accessibility improvements', just to avoid the perception that these specific changes will make mautic fully accessible. They are good additions and do help solve some specific WCAG criteria but not everything all at once.  
@Mike-Dropsolid What do you think?

**Comment by andersonjeccel** [comment]
@LordRembo Oh, GitHub uses this way to call their Accessibility tab, would it be an issue to keep?

image

**Comment by andersonjeccel** [comment]
@escopecz I added that fix for keyboard shortcuts here

**Comment by LordRembo** [comment]
@andersonjeccel @Mike-Dropsolid I had some time to think about this and I think having an separate option to improve Accessibility doesn't make sense at the moment. You need these conditions for it to work:

- Mautic is made WCAG 2.2 AA compliant out of the box
- The extra options to enable, are factors that contribute to AAA classification or otherwise enhance an already accessible site.
- Both things can happen in parallel but the AA state should be priority

With this in mind, I think this PR should take a back-seat to making the site AA compatible, which starts with doing an Accessibility review. Because if the baseline is not good, you can't really enhance it. You're just applying random bandages.  
I would suggest focusing on things like: keyboard navigation, ARIA patterns for all things Ajax and JS, AA contrast rules and semantic html for form inputs. 

That said, the list of extra options in the Accessibility tab looks okay for now. I would maybe rethink the 'link underline' option once the site is AA compliant but that's future work.
If you want to keep working on these enhancements list, I would suggest first making a list of the best options to add. Eg. by going through the WCAG list and look for AAA items. Make a list of options that would implement things that align with those rules.
One thing I'm immediately thinking of, is adding a [higher contrast option](https://www.w3.org/WAI/WCAG22/Understanding/contrast-enhanced). That will also allow you to have a bit of color on this page by default, instead of using black.

File: app/bundles/CoreBundle/Resources/views/Modal/keyboard_shortcuts.html.twig [hunk 1]
```diff
@@ -16,78 +10,78 @@
...
+                  {% set keys = part|split('+') %}
+                  {% for k in keys %}
+                  {% set title = '' %}
+                  {% if k == '⌫' %}
+                  {% set title = 'mautic.keyboard.backspace'|trans %}
+                  {% elseif k == '⌘' %}
+                  {% set title = 'mautic.keyboard.command'|trans %}
+                  {% endif %}
+                  <kbd{% if title %} title="{{ title }}" data-toggle="tooltip" {% endif %}>{{ k }}</kbd>
+                    {% endfor %}
```
Comment (shinde-rahul) [inline app/bundles/CoreBundle/Resources/views/Modal/keyboard_shortcuts.html.twig]: The CS, indentation is missing here.

File: app/bundles/UserBundle/Resources/views/Profile/index.html.twig [hunk 2]
```diff
@@ -37,57 +39,193 @@
...
-                        {{ permissions.editUsername ? form_row(userForm.username) : form_row(userForm.username_unbound) }}
-                        {{ permissions.editEmail ? form_row(userForm.email) : form_row(userForm.email_unbound) }}
-                        {{ form_row(userForm.plainPassword.password) }}
-                        {{ form_row(userForm.plainPassword.confirm) }}
+                    <div class="row">
+                        <div class="col-md-6">
+                            <h3 id="personal-info" class="pt-20">{% trans %}mautic.user.config.title.get_to_know_you{%
+                                endtrans %}</h3>
+                            <div class="text-muted small description pb-20">{% trans
+                                %}mautic.user.config.personalize.interface.helper{% endtrans %}</div>
```
Comment (shinde-rahul) [inline app/bundles/UserBundle/Resources/views/Profile/index.html.twig]:
```suggestion
                            <h3 id="personal-info" class="pt-20">{{ 'mautic.user.config.title.get_to_know_you'|trans }}</h3>
                            <div class="text-muted small description pb-20">{{ 'mautic.user.config.personalize.interface.helper' }}</div>
```

File: app/bundles/UserBundle/Resources/views/Profile/index.html.twig [hunk 3]
```diff
@@ -37,57 +39,193 @@
...
+                            {{ permissions.editName ? form_row(userForm.lastName) : form_row(userForm.lastName_unbound)
+                            }}
+                            {{ permissions.editPosition ? form_row(userForm.position) :
+                            form_row(userForm.position_unbound)
+                            }}
+                            {{ form_row(userForm.signature) }}
+
+                            <h3 id="locale" class="pt-20">{% trans %}mautic.user.config.title.locale{% endtrans %}</h3>
+                            <div class="text-muted small description pb-20">{% trans
+                                %}mautic.user.config.region.preferences.helper{% endtrans %}</div>
```
Comment (shinde-rahul) [inline app/bundles/UserBundle/Resources/views/Profile/index.html.twig]:
```suggestion
                            <h3 id="locale" class="pt-20">{{ 'mautic.user.config.title.locale'|trans }}</h3>
                            <div class="text-muted small description pb-20">{{ 'mautic.user.config.region.preferences.helper'|trans }}</div>
```

File: app/bundles/UserBundle/Resources/views/Profile/index.html.twig [hunk 4]
```diff
@@ -37,57 +39,193 @@
...
+                            {{ form_row(userForm.timezone) }}
+                            {{ form_row(userForm.locale) }}
+                        </div>
+                        <div class="col-md-6">
+                            <h3 id="security" class="pt-20">{% trans %}mautic.user.config.title.account_data_security{%
+                                endtrans %}</h3>
+                            <div class="text-muted small description pb-20">{% trans
+                                %}mautic.user.config.account.security.helper{% endtrans %}</div>
```
Comment (shinde-rahul) [inline app/bundles/UserBundle/Resources/views/Profile/index.html.twig]:
```suggestion
                            <h3 id="security" class="pt-20">{{ 'mautic.user.config.title.account_data_security'|trans }</h3>
                            <div class="text-muted small description pb-20">{{ 'mautic.user.config.account.security.helper'|trans }}</div>
```

**Review by shinde-rahul — CHANGES_REQUESTED** [review]
There are few Twig indentation related issues, that I see. 
What is the best and core style to add translation in twig. That will improve the DX as well.

File: app/bundles/CoreBundle/Resources/views/Modal/keyboard_shortcuts.html.twig (hunk 1 above)
Comment (andersonjeccel) [reply app/bundles/CoreBundle/Resources/views/Modal/keyboard_shortcuts.html.twig]: Fixed!

I always have this issue with the formatter in VS Code, do you know any extension that can properly format HTML with Twig?

**Comment by andersonjeccel** [comment]
@LordRembo That's great to know, thanks for reviewing!

I've been working on small improvement for accessibility across the entire UI over the time (as part of other changes)

One example is converting the right sidebar into a dropdown, it already made the workflow better for keyboard navigation when compared to the previous menu, with appropriate focus indication for each option while using JS libraries from Bootstrap

I think Ruth is waiting for an organization to provide an accessibility audit, that can be something to look in the future (but really far from now, no plans to tackle more of these)

**Review by RCheesley — APPROVED** [review]
Thanks for the work on this PR.

I do hear what @LordRembo is saying about having a clear plan as to what is needed to really make Mautic accessible, however I also am pleased to see we're at least making a start with some basic changes which enable folks to turn off some of the whizzy things we recently introduced, if they want.

It also allows us to open the discussion on the importance of accessibility and hopefully to bring in more people with ideas and thoughts on how we can further develop this important aspect. I'd rather we make small, iterative steps than wait until we have the 'big hurrah' project ready to go.

I had been in touch with an agency about having a formal audit which we could then pick apart into chunks and address over time, but we didn't get the funding from NLNet - I think mainly because we didn't have enough detail before the call for submissions closed - but it's perhaps something we can further explore. I agree we should have a 'north star' to use the horrible corporate lingo, which we're orienting towards as far as accessibility, and then chunk that down into actionable steps which we can perhaps plan to tag onto the release timelines. I've re-opened the conversation to see what support we might be eligible for as far as probono support goes.

File: app/bundles/CoreBundle/Resources/views/Modal/keyboard_shortcuts.html.twig (hunk 1 above)
Comment (shinde-rahul) [reply app/bundles/CoreBundle/Resources/views/Modal/keyboard_shortcuts.html.twig]: @andersonjeccel, I use PHPStrom, and it does help with indentation.

Google said this https://codeknight.co.uk/blog/getting-prettier-working-with-twig-craft-cms to be useful, 🤞

**Pull request merged**